modules/core/role/src/Plugin/ScopeGranularity/ManagedRolePermissions.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\farm_role\Plugin\ScopeGranularity;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+use Drupal\farm_role\ManagedRolePermissionsManagerInterface;
+use Drupal\simple_oauth\Attribute\ScopeGranularity;
+use Drupal\simple_oauth\Plugin\ScopeGranularity\Role;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * The Managed Role Permissions scope granularity plugin.
+ */
+#[ScopeGranularity(
+  'managed_role_permissions',
+  new TranslatableMarkup('Managed Role Permissions'),
+)]
```

Comment:
I'm not sure what is the best naming for this scope granularity plugin. Developers (static scopes) and Users (scope config entities via UI) just need to know that they need to use a "Managed Role Permissions" granularity instead of "Role" granularity if they want any of the _role's managed permissions_ to be considered when authorizing via oauth.

Reply:
I've pushed a commit that changes this.